sorted_db/db.go

```diff
@@ -92,17 +101,18 @@ func (db *DB) close() {
 		db.f = nil
 	}
 	db.size = -1
+	return
 }
 
 // Remap DB to the same backing file
 func (db *DB) Remap() error {
-	db.RLock()
+	db.mutex.RLock()
```

Comment:
Also personal coding style, I tend to avoid `defer` for hot functions when the function is also short and you don't loose much readability. While `defer` is cheap, it's not free.  That's why `Search`, `ForwardMatch` and `RangeMatch` don't use it.